### PR TITLE
CORE-436 Report the amount of disk consumption being utilized by RSpace

### PR DESCRIPTION
**Pull request by ys-pyrofex**
## Overview
This PR adds RSpace disk consumption metrics to the GRPC service and Prometeus chart.

### Does this PR relate to an RChain JIRA issue? 
CORE-436

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/grpc.scala [hunk 1]
```diff
@@ -22,15 +22,16 @@ import monix.execution.Scheduler
...
 import coop.rchain.rholang.interpreter.errors.InterpreterError
 
 object GrpcServer {
 
   def acquireServer[
       F[_]: Capture: Monad: MultiParentCasper: NodeDiscovery: JvmMetrics: NodeMetrics: Futurable](
       port: Int,
-      runtime: Runtime)(implicit scheduler: Scheduler): F[Server] =
+      runtime: Runtime)(implicit scheduler: Scheduler): F[Server] = {
+    implicit val storeMetricsCapture: StoreMetrics[F] = diagnostics.storeMetrics[F](runtime.store)
```
Comment (sebtomba) [inline node/src/main/scala/coop/rchain/node/grpc.scala]: Can you move this to **node.scala** and add `StoreMetrics` to `F[_]]` capabilieties?

File: rspace/src/main/scala/coop/rchain/rspace/IStore.scala [hunk 2]
```diff
@@ -4,6 +4,8 @@ import coop.rchain.rspace.internal._
 
 import scala.collection.immutable.Seq
 
+case class StoreSize(sizeOnDisk: Long, dataEntries: Long)
```
Comment (henrytill) [inline rspace/src/main/scala/coop/rchain/rspace/IStore.scala]: Could we move this elsewhere?  I'd like this file to only contain the definition of `IStore`.  I'm fine with putting this in its own file for now.

File: rspace/src/main/scala/coop/rchain/rspace/LMDBStore.scala [hunk 3]
```diff
@@ -20,7 +20,8 @@ import scala.collection.immutable.Seq
   *
   * To create an instance, use [[LMDBStore.create]].
   */
-class LMDBStore[C, P, A, K] private (val env: Env[ByteBuffer],
+class LMDBStore[C, P, A, K] private (env: Env[ByteBuffer],
+                                     databaseBath: Path,
```
Comment (henrytill) [inline rspace/src/main/scala/coop/rchain/rspace/LMDBStore.scala]: 🛀 🛀 🛀 

I assume you meant `databasePath`?

File: rspace/src/main/scala/coop/rchain/rspace/util/package.scala [hunk 4]
```diff
@@ -1,5 +1,7 @@
 package coop.rchain.rspace
 
+import java.util.Locale
```
Comment (henrytill) [inline rspace/src/main/scala/coop/rchain/rspace/util/package.scala]: unused

File: shared/src/main/scala/coop/rchain/shared/LongOps.scala [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+object LongOps {
+
+  /**
+    * Formats size in bytes to human-readable format
+    * Avoids use of slow math operations and loops
+    *
+    * Assumes 1 KB = 1024 bytes, not 1000 bytes as some other methods do
+    */
+  def toHumanReadableSize(value: Long): String =
```
Comment (henrytill) [inline shared/src/main/scala/coop/rchain/shared/LongOps.scala]: My take on the `<name>Ops` objects is that they should contain [implicit classes](https://docs.scala-lang.org/overviews/core/implicit-classes.html) that enrich existing types.  

So `LongOps` here would contain:

```scala
  implicit class RichLong(value: Long) {

    /**
      * Formats size in bytes to human-readable format
      * Avoids use of slow math operations and loops
      *
      * Assumes 1 KB = 1024 bytes, not 1000 bytes as some other methods do
      */
    def toHumanReadableSize: String =
      if (value < 1024) {
        value + " B"
      } else {
        val z = 63 - numberOfLeadingZeros(value)
        "%.1f %sB".formatLocal(Locale.US, value.toDouble / (1L << z), " KMGTPE".charAt(z / 10))
      }
  }
```

When `LongOps` is brought into scope, you could then use it like:

```scala
  def showStoreUsage(storeUsage: StoreUsage): String =
    s"""Store metrics:
       | - Size On Disk: ${storeUsage.sizeOnDisk.toHumanReadableSize}
       | - Data Entries: ${storeUsage.dataEntries}
       |""".stripMargin
```

File: node/src/main/scala/coop/rchain/node/grpc.scala (hunk 1 above)
Comment (ys-pyrofex) [reply node/src/main/scala/coop/rchain/node/grpc.scala]: Sure, updated.

**Comment by henrytill** [comment]
@sebtomba: Are you handling the `node` side of this review?  I'm not really equipped to review the correctness of the things over there...

**Comment by sebtomba** [comment]
@henrytill besides your requested changes LGTM

File: rspace/src/main/scala/coop/rchain/rspace/IStore.scala (hunk 2 above)
Comment (ys-pyrofex) [reply rspace/src/main/scala/coop/rchain/rspace/IStore.scala]: I don't think that it's good idea to separate interface and the "return value". Anyway extract it (and return back if needed) - just a single mouse click. Done.

File: rspace/src/main/scala/coop/rchain/rspace/LMDBStore.scala (hunk 3 above)
Comment (ys-pyrofex) [reply rspace/src/main/scala/coop/rchain/rspace/LMDBStore.scala]: Sure :))))

File: shared/src/main/scala/coop/rchain/shared/LongOps.scala (hunk 5 above)
Comment (ys-pyrofex) [reply shared/src/main/scala/coop/rchain/shared/LongOps.scala]: Thanks for a tip! Here's the place where all my theoretical knowledge from scala books lost somewhere when they're needed in practice.

**Pull request merged**